demo/src/declerations.d.ts

```diff
@@ -0,0 +1,5 @@
+// Prevents module not found errors when importing images
```

Comment:
This is good, but name the file `index.d.ts`. Note that the original file name you used was misspelled (should be "declarations" not "declerations")

Reply:
Also, you should integrate this image loading with our webpack configuration: https://webpack.js.org/guides/asset-management/#loading-images.

Reply:
1. I've integrated the image loading to the webpack configuration and it works properly.
2. As for renaming the file to `index.d.ts`, once I rename the file, the module not found error reemerges. I did some digging on Stack Overflow and I found [this post](https://stackoverflow.com/questions/59728371/typescript-d-ts-file-not-recognized) which suggests that you can't have both `index.ts` and `index.d.ts` (please see screenshot). That being said, when I went through the documentation, I was not able to directly locate the content quoted in the answer. I'm also not sure if this extends to `index.tsx`, but it seems that the `.d.ts` file works with any filename other than `index.d.ts`. I wanted to ask if I'm missing something in terms of configuring other files in the repository (such as `tsconfig.json`)?

<details>
<summary>Stack Overflow</summary>

![image](https://github.com/user-attachments/assets/bbd3034d-09f1-48e2-9b59-4a03c342c621)


</details>